testutil/terraform_cache.go

```diff
@@ -0,0 +1,185 @@
+//go:build linux || darwin
+
+package testutil
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	// TerraformConfigFileName is the name of the Terraform CLI config file.
+	TerraformConfigFileName = "terraform.rc"
+	// CacheProvidersDirName is the subdirectory name for the provider mirror.
+	CacheProvidersDirName = "providers"
+	// CacheTemplateFilesDirName is the subdirectory name for template files.
+	CacheTemplateFilesDirName = "files"
+)
+
+// HashTemplateFilesAndTestName generates a unique hash based on test name and template files.
+func HashTemplateFilesAndTestName(t *testing.T, testName string, templateFiles map[string]string) string {
```

Comment:
If we're not going to be using these functions externally (and don't plan to in the future) then we should just keep them private for now. It seems you're only using `GetTestCacheDir` and `CacheProviders` externally